pkg/virtual/workspaces/registry/rest.go

```diff
@@ -0,0 +1,482 @@
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return workspaceList, nil
+}
+
+var _ = rest.Getter(&REST{})
+
+// Get retrieves a Workspace by name
+func (s *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	opts := metav1.GetOptions{}
+	if options != nil {
+		opts = *options
+	}
+
+	user, ok := apirequest.UserFrom(ctx)
+	if !ok {
+		return nil, kerrors.NewForbidden(tenancyv1alpha1.Resource("workspace"), "", fmt.Errorf("unable to list workspaces without a user on the context"))
+	}
+
+	if scope := ctx.Value(WorkspacesScopeKey); scope == personalScope {
+		internalName, err := s.getInternalNameFromPrettyName(user, name)
+		if err != nil {
+			return nil, err
+		}
+		name = internalName
+	}
+
+	workspace, err := s.workspaceClient.Get(ctx, name, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if scope := ctx.Value(WorkspacesScopeKey); scope == personalScope {
+		workspace.Name, err = s.getPrettyNameFromInternalName(user, workspace.Name)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// TODO Filtering by applying the List operation would probably not be necessary anymore
+	// when using a semi-delegated authorizer in the workspaces virtual workspace that would
+	// delegate this authorization to the main KCP instance hosting the workspaces and RBAC rules
+	obj, err := s.List(ctx, &metainternal.ListOptions{
+		ResourceVersion: opts.ResourceVersion,
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, ws := range obj.(*tenancyv1alpha1.WorkspaceList).Items {
+		if ws.Name == workspace.Name && ws.ClusterName == workspace.ClusterName {
+			return workspace, nil
+		}
+	}
+
+	return nil, kerrors.NewNotFound(tenancyv1alpha1.SchemeGroupVersion.WithResource("workspaces").GroupResource(), name)
+}
+
+type RoleType string
+
+const (
+	ListerRoleType RoleType = "lister"
+	AdminRoleType  RoleType = "admin"
+)
+
+var roleRules map[RoleType]rbacv1.PolicyRule = map[RoleType]rbacv1.PolicyRule{
+	ListerRoleType: {
+		Verbs:     []string{"get"},
+		Resources: []string{"workspaces"},
+	},
+	AdminRoleType: {
+		Verbs:     []string{"get", "delete"},
+		Resources: []string{"workspaces"},
+	},
+}
+
+func getRoleBindingName(roleType RoleType, workspacePrettyName string, user kuser.Info) string {
+	return string(roleType) + "-workspace-" + workspacePrettyName + "-" + user.GetName()
+}
+
+func InternalListOptionsToSelectors(options *metainternal.ListOptions) (labels.Selector, fields.Selector) {
+	label := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		label = options.LabelSelector
+	}
+	field := fields.Everything()
+	if options != nil && options.FieldSelector != nil {
+		field = options.FieldSelector
+	}
+	return label, field
+}
+
+var _ = rest.Creater(&REST{})
+
+// Create creates a new version of a resource.
+func (s *REST) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
```

Comment:
can go doc the steps, with a reasoning for the order? As we did in the gdoc.

Reply:
Added Go doc at the beginning of the function